File: packages/ruby-sdk/lib/strait/operations/event_triggers.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for event trigger management and dispatching.
    class EventTriggersService < BaseService
      # List all event trigger events.
      # GET /v1/event-triggers/events
      def list_events(query: nil)
        _request(:get, "/v1/event-triggers/events", query: query)
      end

      # Get an event by key.
      # GET /v1/event-triggers/events/{eventKey}
      def get_event(event_key)
        _request(:get, "/v1/event-triggers/events/{eventKey}", path_params: { "eventKey" => event_key })
      end

      # Delete an event by key.
      # DELETE /v1/event-triggers/events/{eventKey}
      def delete_event(event_key)
        _request(:delete, "/v1/event-triggers/events/{eventKey}", path_params: { "eventKey" => event_key })
        nil
      end

      # Send an event by key.
      # POST /v1/event-triggers/events/{eventKey}/send
      def send_event(event_key, body)
        _request(:post, "/v1/event-triggers/events/{eventKey}/send", path_params: { "eventKey" => event_key }, body: body)
      end

      # Send events matching a prefix.
      # POST /v1/event-triggers/prefix/{prefix}/send
      def send_prefix(prefix, body)
        _request(:post, "/v1/event-triggers/prefix/{prefix}/send", path_params: { "prefix" => prefix }, body: body)
      end

      # Purge event triggers.
      # POST /v1/event-triggers/purge
      def purge_event(body)
        _request(:post, "/v1/event-triggers/purge", body: body)
      end

      # Get event trigger statistics.
      # GET /v1/event-triggers/stats
      def get_stat
        _request(:get, "/v1/event-triggers/stats")
      end
    end
  end
end

File: packages/python-sdk/strait/operations/event_triggers.py
"""Event trigger management operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class EventTriggersService(BaseService):
    def list_events(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/events", query=query)

    def get_event(self, event_key: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/events/{eventKey}", path_params={"eventKey": event_key},
        )

    def delete_event(self, event_key: str) -> dict[str, Any]:
        return self._request(
            "DELETE", "/v1/events/{eventKey}", path_params={"eventKey": event_key},
        )

    def send_event(self, event_key: str, body: Any) -> dict[str, Any]:
        return self._request(
            "POST", "/v1/events/{eventKey}/send",
            path_params={"eventKey": event_key}, body=body,
        )

    def get_stream(self, event_key: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/events/{eventKey}/stream", path_params={"eventKey": event_key},
        )

    def send_prefix(self, prefix: str, body: Any) -> dict[str, Any]:
        return self._request(
            "POST", "/v1/events/prefix/{prefix}/send",
            path_params={"prefix": prefix}, body=body,
        )

    def purge_event(self, body: Any) -> dict[str, Any]:
        return self._request("POST", "/v1/events/purge", body=body)

    def get_stat(self) -> dict[str, Any]:
        return self._request("GET", "/v1/events/stats")


class AsyncEventTriggersService(AsyncBaseService):
    async def list_events(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/events", query=query)

    async def get_event(self, event_key: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/events/{eventKey}", path_params={"eventKey": event_key},
        )

    async def delete_event(self, event_key: str) -> dict[str, Any]:
        return await self._request(
            "DELETE", "/v1/events/{eventKey}", path_params={"eventKey": event_key},
        )

    async def send_event(self, event_key: str, body: Any) -> dict[str, Any]:
        return await self._request(
            "POST", "/v1/events/{eventKey}/send",
            path_params={"eventKey": event_key}, body=body,
        )

    async def get_stream(self, event_key: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/events/{eventKey}/stream", path_params={"eventKey": event_key},
        )

    async def send_prefix(self, prefix: str, body: Any) -> dict[str, Any]:
        return await self._request(
            "POST", "/v1/events/prefix/{prefix}/send",
            path_params={"prefix": prefix}, body=body,
        )

    async def purge_event(self, body: Any) -> dict[str, Any]:
        return await self._request("POST", "/v1/events/purge", body=body)

    async def get_stat(self) -> dict[str, Any]:
        return await self._request("GET", "/v1/events/stats")
